pennylane/transforms/dynamic_one_shot.py

```diff
@@ -206,20 +206,13 @@ def init_auxiliary_tape(circuit: qml.tape.QuantumScript):
                 new_measurements.append(SampleMP(obs=m.obs))
             else:
                 new_measurements.append(m)
-    new_operations = []
     for op in circuit.operations:
-        if "MidCircuitMeasure" in str(type(op)):  # pragma: no cover
-            new_op = op
-            new_op.bypass_postselect = True
-            new_operations.append(new_op)
-        else:
-            new_operations.append(op)
         if "MidCircuitMeasure" in str(type(op)):  # pragma: no cover
             new_measurements.append(qml.sample(op.out_classical_tracers[0]))
         elif isinstance(op, MidMeasureMP):
             new_measurements.append(qml.sample(MeasurementValue([op], lambda res: res)))
```

Comment:
Just had a realization. We could store the measurement value/tracer in `MidMeasureMP.mv` once `MidCircuitMeasure` inherits `MidMeasureMP` and then we can get rid of this branch. Not needed right now, but I can come back and update this once the Catalyst side work is complete. Then this line could instead be the following, and we won't need another branch for `MidMeasureMP`.
```suggestion
        if isinstance(op, MidMeasureMP):
            new_measurements.append(qml.sample(op=op.mv))
```

```suggestion
        if "MidCircuitMeasure" in str(type(op)):  # pragma: no cover
            new_measurements.append(qml.sample(op.out_classical_tracers[0]))
        elif isinstance(op, MidMeasureMP):
            new_measurements.append(qml.sample(MeasurementValue([op], lambda res: res)))
```

Reply:
This change will not work right now though.